lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -179,7 +180,13 @@ async def _sync_subscribers(self) -> bool:
                 self._store.upsert_subscriber(subscriber_data)
             for sid in changeset.deleted:
                 self._store.delete_subscriber(sid)
-            self._detach_subscribers_by_ids(changeset.deleted)
+
+            inactive = [
+                SIDUtils.to_str(subscriber.sid)
+                for subscriber in changeset.to_renew
+                if (subscriber.lte and subscriber.lte.state != LTESubscription.ACTIVE)
+            ]
+            self._detach_subscribers_by_ids(list(changeset.deleted) + inactive)
```

Comment:
`changeset.deleted` should already be a list, so we don't need to wrap it in another `list(...)`?

Reply:
The typing of the protobuf objects is a little weird. `changeset.deleted` is actually a RepeatedScalarContainer and you get a runtime error if you try to add elements without converting into a list.

Reply:
Interesting!